docs/source/configuration/authn-subgraph.mdx

```diff
@@ -0,0 +1,56 @@
+---
+title: Subgraph Authentication in the Apollo Router
+---
+
+The Apollo Router supports subgraph request authentication and key rotation via [AWS Signature Version 4](https://docs.aws.amazon.com/AmazonS3/latest/API/sig-v4-authenticating-requests.html).
+
+**To use this feature:**
+
+- You must have properly set up your AWS hosted subgraphs with IAM, so they accept [signed requests](https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_aws-signing.html).
+
+## How it works
+
+Subgraph requests are signed using [HTTP Authorization headers](https://docs.aws.amazon.com/AmazonS3/latest/API/sigv4-auth-using-authorization-header.html)
+
+### Configuration options
+
+
+#### Full example
+
+Below is a full example with a default credentials chain for all subgraphs, and with hardcoded credentials for the `products` subgraph:
+
+    ```yaml title="router.yaml"
+    authentication:
+      subgraph:
+        all: # configuration that will apply to all subgraphs
+          aws_sig_v4:
+            default_chain:
+              profile_name: "my-test-profile" # https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/iam-roles-for-amazon-ec2.html#ec2-instance-profile
+              region: "us-east-1" # https://docs.aws.amazon.com/general/latest/gr/rande.html
+              service_name: "lambda" # https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_aws-services-that-work-with-iam.html
+              assume_role: # https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles.html
+                role_arn: "test-arn"
+                session_name: "test-session"
+                external_id: "test-id"
+        subgraphs:
+          products:
+            aws_sig_v4:
+              hardcoded: # Not recommended, prefer using default_chain as shown above
+                access_key_id: "my-access-key"
+                secret_access_key: "my-secret-access-key"
+                region: "us-east-1"
+                service_name: "vpc-lattice-svcs" # "s3", "lambda" etc.
+    ```
+
+
+#### Default chain:
+
+The default chain authentication method will try to resolve credentials in the following order:
+  - Environment variables (`AWS_ACCESS_KEY_ID`, `AWS_SECRET_ACCESS_KEY` or `SECRET_ACCESS_KEY`, `AWS_SESSION_TOKEN`, `AWS_ROLE_ARN`, `AWS_IAM_ROLE_SESSION_NAME`)
```

Comment:
This might be more readable as a table:

```suggestion

| Credential Type                  | Examples                                                                                             |
|----------------------------------|------------------------------------------------------------------------------------------------------|
| Environment variables            | `AWS_ACCESS_KEY_ID`, `AWS_SECRET_ACCESS_KEY` or `SECRET_ACCESS_KEY`, `AWS_SESSION_TOKEN`, `AWS_ROLE_ARN`, `AWS_IAM_ROLE_SESSION_NAME`|
| Shared configurations             | `~/.aws/config`, `~/.aws/credentials`, configured with `AWS_CONFIG_FILE` and `AWS_SHARED_CREDENTIALS_FILE` environment variables           |
| Web identity tokens                           | Possibly configured with the `AWS_WEB_IDENTITY_TOKEN_FILE` environment variable            | `AWS_WEB_IDENTITY_TOKEN_FILE`
| Elastic Container Service (ECS)                           | Possibly configured with the `AWS_CONTAINER_CREDENTIALS_RELATIVE_URI`, `AWS_CONTAINER_CREDENTIALS_FULL_URI`, and `AWS_CONTAINER_AUTHORIZATION_TOKEN` environment variables, in that order |
```

Reply:
If you like the table idea, best to make the order clear in the preceding sentence:


The default chain authentication method tries to resolve credentials in the following order, starting with environment variables:

Reply:
love it!

Reply:
Fixed in [d6b6aa0](https://github.com/apollographql/router/pull/3365/commits/d6b6aa07478e5f41f65b51a8e29cb8e06b8f1f4a)